### PR TITLE
Add new data source: google_dns_managed_zone

### LINKED ISSUE #239: Support google_dns_managed_zone as a data source
DNS needs to be created out of band at some point. Support importing a Cloud DNS zone so the user can provision a domain/subdomain and setup a cut to the zone out of band, then use terraform to manage the records. 

### Terraform Version
0.9.11

### Affected Resource(s)
- google_dns_managed_zone

### Terraform Configuration Files
```hcl
data "google_dns_managed_zone" "env_dns_zone" {
  name        = "${var.dns_zone_name}"
  dns_name    = "${var.env_name}.${var.dns_suffix}."
  description = "DNS zone for the ${var.env_name} environment"
}

resource "google_dns_record_set" "dns" {
  name = "my-address.${data.google_dns_managed_zone.env_dns_zone.dns_name}"
  type = "A"
  ttl  = 300

  managed_zone = "${data.google_dns_managed_zone.env_dns_zone.name}"

  rrdatas = ["${google_compute_instance.exposed-instance.network_interface.0.address}"]
}
```

### PR DESCRIPTION
**Pull request by rosbo**
Fixes #239 

cc/ @johnsonj @danawillow

### CODE REVIEW

File: google/data_source_dns_managed_zone_test.go [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+		dsFullName := "data.google_dns_managed_zone." + dsName
+		rsFullName := "google_dns_managed_zone." + rsName
+		ds, ok := s.RootModule().Resources[dsFullName]
+		if !ok {
+			return fmt.Errorf("root module has no resource called %s", dsFullName)
+		}
+
+		rs, ok := s.RootModule().Resources[rsFullName]
+		if !ok {
+			return fmt.Errorf("can't find %s in state", rsFullName)
```
Comment (danawillow) [inline google/data_source_dns_managed_zone_test.go]: Any reason to have different wording for the two error messages?

File: website/docs/d/dns_managed_zone.html.markdown [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+---
+
+# google\_dns\_managed\_zone
+
+Provides access to a zone's attributes within Google Cloud DNS.
+For more information see
+[the official documentation](https://cloud.google.com/dns/zones/)
+and
+[API](https://cloud.google.com/dns/api/v1/managedZones).
+```
```
Comment (danawillow) [inline website/docs/d/dns_managed_zone.html.markdown]: This should probably be ```tf or ```hcl

File: website/docs/d/dns_managed_zone.html.markdown [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+  managed_zone = "${data.google_dns_managed_zone.env_dns_zone.name}"
+
+  rrdatas = ["test"]
+}
+```
+
+## Argument Reference
+
+* `name` - (Required) A unique name for the resource, required by GCE.
```
Comment (danawillow) [inline website/docs/d/dns_managed_zone.html.markdown]: GCE isn't relevant here, is it?

File: website/docs/d/dns_managed_zone.html.markdown [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    Changing this forces a new resource to be created.
```
Comment (danawillow) [inline website/docs/d/dns_managed_zone.html.markdown]: Not for a data source

File: website/docs/d/dns_managed_zone.html.markdown [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+* `project` (optional) - ID of the project to list available cluster versions for. Should match the project the cluster will be deployed to.
```
Comment (danawillow) [inline website/docs/d/dns_managed_zone.html.markdown]: This looks like it needs to be updated from what this page was copied from

File: website/docs/d/dns_managed_zone.html.markdown [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+  Defaults to the project that the provider is authenticated with.
+
+## Attributes Reference
+
+The following attributes are exported:
+
+* `dns_name` - The DNS name of this zone, e.g. "terraform.io".
+
+* `description` - A textual description field. Defaults to 'Managed by Terraform'.
```
Comment (danawillow) [inline website/docs/d/dns_managed_zone.html.markdown]: Defaults aren't really relevant for data sources

File: google/data_source_dns_managed_zone_test.go (hunk 1 above)
Comment (rosbo) [inline google/data_source_dns_managed_zone_test.go]: Other than that I copied and pasted this snippet from another test. No ;) 

Fixed

File: website/docs/d/dns_managed_zone.html.markdown (hunk 2 above)
Comment (rosbo) [inline website/docs/d/dns_managed_zone.html.markdown]: Fixed

File: website/docs/d/dns_managed_zone.html.markdown (hunk 3 above)
Comment (rosbo) [inline website/docs/d/dns_managed_zone.html.markdown]: Oupps. I meant to clean up these descriptions before opening the PR... Forgot about it ;)

File: website/docs/d/dns_managed_zone.html.markdown (hunk 4 above)
Comment (rosbo) [inline website/docs/d/dns_managed_zone.html.markdown]: Fixed

File: website/docs/d/dns_managed_zone.html.markdown (hunk 5 above)
Comment (rosbo) [inline website/docs/d/dns_managed_zone.html.markdown]: Fixed

File: website/docs/d/dns_managed_zone.html.markdown (hunk 6 above)
Comment (rosbo) [inline website/docs/d/dns_managed_zone.html.markdown]: Fixed

**Comment by danawillow** [comment]
```
make testacc TEST=./google TESTARGS='-run=TestAccDataSourceDnsManagedZone'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./google -v -run=TestAccDataSourceDnsManagedZone -timeout 120m
=== RUN   TestAccDataSourceDnsManagedZone_basic
--- PASS: TestAccDataSourceDnsManagedZone_basic (3.36s)
PASS
ok  	github.com/terraform-providers/terraform-provider-google/google	3.516s
```

LGTM, merge at will

**Comment by johnsonj** [comment]
Thank you!

**Pull request merged**